libs/pages/services/src/lib/feature/page-application-create-feature/page-application-create-port-feature/page-application-create-port-feature.tsx

```diff
@@ -0,0 +1,105 @@
+import { useEffect, useState } from 'react'
+import { FormProvider, useForm } from 'react-hook-form'
+import { useNavigate, useParams } from 'react-router'
+import {
+  SERVICES_APPLICATION_CREATION_URL,
+  SERVICES_CREATION_GENERAL_URL,
+  SERVICES_CREATION_INSTALLATION_URL,
+  SERVICES_CREATION_RESOURCES_URL,
+  SERVICES_URL,
+} from '@qovery/shared/router'
+import { FunnelFlowBody, FunnelFlowHelpCard } from '@qovery/shared/ui'
+import PageApplicationCreatePort from '../../../ui/page-application-create/page-application-create-port/page-application-create-port'
+import { PortData } from '../application-creation-flow.interface'
+import { useApplicationContainerCreateContext } from '../page-application-create-feature'
+
+export function PageApplicationCreatePortFeature() {
+  const { setCurrentStep, portData, setPortData, generalData } = useApplicationContainerCreateContext()
+  const { organizationId = '', projectId = '', environmentId = '' } = useParams()
+  const navigate = useNavigate()
+
+  useEffect(() => {
+    !generalData?.name &&
+      navigate(
+        `${SERVICES_URL(organizationId, projectId, environmentId)}${SERVICES_APPLICATION_CREATION_URL}` +
+          SERVICES_CREATION_GENERAL_URL
+      )
+  }, [generalData, navigate, environmentId, organizationId, projectId])
+
+  const funnelCardHelp = (
+    <FunnelFlowHelpCard
+      title="Configuring the application port"
+      items={[
+        'Declare the ports used internally by your application',
+        'Declared ports are accessible from other applications within the same environment',
+        'You can also expose them on the internet by making them public.',
+        'Declared ports are also used to check the liveness/readiness of your application.',
+      ]}
+      helpSectionProps={{
+        description: 'This is still a description',
+        links: [
+          {
+            link: 'https://hub.qovery.com/docs/using-qovery/configuration/application/#ports',
+            linkLabel: 'How to configure my application',
+            external: true,
+          },
+          { link: 'https://discuss.qovery.com/', linkLabel: 'Still need help? Ask on our Forum', external: true },
+        ],
+      }}
+    />
+  )
+
+  useEffect(() => {
+    setCurrentStep(3)
+  }, [setCurrentStep])
+
+  const methods = useForm<PortData>({
+    defaultValues: portData,
+    mode: 'onChange',
+  })
+
+  const onSubmit = methods.handleSubmit((data) => {
+    setPortData(data)
+    const pathCreate = `${SERVICES_URL(organizationId, projectId, environmentId)}${SERVICES_APPLICATION_CREATION_URL}`
+    navigate(pathCreate + SERVICES_CREATION_INSTALLATION_URL)
+  })
+
+  const onBack = () => {
+    const pathCreate = `${SERVICES_URL(organizationId, projectId, environmentId)}${SERVICES_APPLICATION_CREATION_URL}`
```

Comment:
Why you didn't put it outside the function `const pathCreate`? You use it on the "submit" and "back"